core/src/modules/ops/source.ts

```diff
@@ -554,7 +554,7 @@ export namespace SourceOps {
       retry = false;
 
       const property = Property.build({
-        id: id ?? ConfigWriter.generateId(key),
+        id,
```

Comment:
So there's never a time id would be null?

Reply:
If an `id` is not passed into `bootstrapUniqueProperty`, then Grouparoo will generate a UUID for the property as usual. Added explicit tests for this here cbed94dd5b8703c65fe661205cbcbdc0130f3581